netam/model_factory.py

```diff
@@ -0,0 +1,350 @@
+"""Model factory functions for creating netam models.
+
+This module provides factory functions to create netam selection models from
+configuration dictionaries and YAML files. Supports all model types including:
+- SingleValueBinarySelectionModel
+- ParentIndependentBinarySelectionModel
+- TransformerBinarySelectionModelLinAct
+- TransformerBinarySelectionModelWiggleAct
+- BidirectionalTransformerBinarySelectionModel
+
+Example usage:
+    >>> import netam
+    >>> config = {'model_class': 'SingleValueBinarySelectionModel',
+    ...           'hparams': {'model_type': 'dasm'}}
+    >>> model = netam.create_selection_model_from_dict(config, torch.device('cpu'))
+"""
+
+import torch
+import inspect
+import yaml
+from typing import Dict, Any, List, Optional
+from pathlib import Path
+
+
+# Path to model configurations directory
+MODEL_CONFIGS_DIR = Path(__file__).parent / "model_configs"
```

Comment:
Is this robust? I thought there was some python facility for a package having data. If this works, that's fine tho.